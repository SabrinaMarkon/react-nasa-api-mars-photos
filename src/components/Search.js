import React, { Component } from 'react';

export default class Search extends Component {

  constructor(props) {
    super(props);
    this.state = {
      searchinput: '',
      searchfield: 'name'
    }
    this.handleChange = this.handleChange.bind(this);
    this.handleSelected = this.handleSelected.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange = event => {
    let searchinput = (event.target.value);
    this.setState({
      searchinput
    });
  }

  handleSelected = event => {
    let searchfield = (event.target.value);
    this.setState({
      searchfield
    });
  }

  handleSubmit = event => {
    event.preventDefault();
    this.props.doSearch(this.state.searchfield, this.state.searchinput);
  }

  render() {
    return (
    <form>
      <select className="form-input" value={this.state.searchfield} onChange={this.handleSelected}>
        <option value="name">Name</option>
        <option value="nametype">Name Type</option>
        <option value="recclass">Rec Class</option>
        {/* <option value="mass">Mass (g)</option> */}
        <option value="fall">Fall</option>
        {/* <option value="year">Year</option> */}
        <option value="reclat">Latitude</option>
        <option value="reclong">Longitude</option>
      </select>
      <input
        className="form-input"
        type="text"
        value={this.state.searchinput}
        onChange={this.handleChange}
      />
      <button className="form-input" type="button" onClick={this.handleSubmit}>Search!</button>
    </form>
    )
  }
}