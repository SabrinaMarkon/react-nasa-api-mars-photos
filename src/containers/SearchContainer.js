import React, { Component } from 'react';
import Search from '../components/Search';

export default class SearchContainer extends Component {

  constructor(props) {
    super(props);
    this.doSearch = this.doSearch.bind(this);
  }

  doSearch = (searchfield, searchinput) => {
    this.props.doSearch(searchfield, searchinput);
  }

  render() {
    return (
      <div className="center-align pt-2">
        <Search doSearch={this.doSearch} />
      </div>
    );
  }
}