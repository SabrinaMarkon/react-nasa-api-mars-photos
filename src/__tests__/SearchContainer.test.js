import React from 'react';
import ReactDOM from 'react-dom';

// components
import SearchContainer from '../containers/SearchContainer';
import Search from '../components/Search';

function setup() {
  const wrapper = shallow(<SearchContainer />);
  return wrapper;
}

describe('Testing the SearchContainer component', () => {
  it('renders the SearchContainer box', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a child Search component', () => {
    const wrapper = shallow(<Search />);
    expect(wrapper.exists()).toBe(true);
  });
});