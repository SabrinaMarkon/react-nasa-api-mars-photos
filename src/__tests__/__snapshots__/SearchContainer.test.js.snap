// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the SearchContainer component renders the SearchContainer box 1`] = `
<div>
  <Search
    doSearch={[Function]}
  />
</div>
`;
